### ISSUE TITLE
Rewrite handler crash when there's no match

### ISSUE THREAD

**Issue by leshik** [tomMoulard/htransformation#33]
When there's no match, the rewrite handler crashes [here](https://github.com/tomMoulard/htransformation/blob/0e8eec39a1beff3140ccc6f26d1bb1c503905bac/pkg/handler/rewrite/rewrite.go#L29) with:
```
reflect.Value.Slice: slice index out of bounds  middlewareName=traefik-internal-recovery middlewareType=Recovery
```

**Comment by tomMoulard** [comment]
Hello @leshik,

Thanks for your interest in this Traefik plugin,

Can you elaborate on the bug you seem to have ?

For this matcher, I've added test on this there : 167b62e357783aef693ac269f362a207f81003a0
https://github.com/tomMoulard/htransformation/blob/0e8eec39a1beff3140ccc6f26d1bb1c503905bac/pkg/handler/rewrite/rewrite_test.go#L56-L68

**Comment by leshik** [comment]
Hi @tomMoulard,

I suppose there might be a missing `continue` inside the `if` block, and `range` then panics on an empty `capture` slice.

**Comment by tomMoulard** [comment]
This should be fixed in 5e973ec83c29f5654d00a768da4621120e3873f1.

Thanks :smiley:

**Comment by leshik** [comment]
@tomMoulard sorry, missed the EDIT. Yes, the fix forks.
